google_guest_agent/snapshot_listener.go

```diff
@@ -0,0 +1,202 @@
+//  Copyright 2020 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	sspb "github.com/GoogleCloudPlatform/guest-agent/google_guest_agent/snapshot_service"
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+	"github.com/golang/groupcache/lru"
+	"google.golang.org/grpc"
+)
+
+var (
+	seenPreSnapshotOperationIds  = lru.New(128)
+	seenPostSnapshotOperationIds = lru.New(128)
+)
+
+type snapshotConfig struct {
+	timeout               int // seconds
+	preSnapshotScriptURL  string
+	postSnapshotScriptURL string
+	enabled               bool
+	snapshotServiceIP     string
+	snapshotServicePort   int
+}
+
+type invalidSnapshotConfig struct {
+	msg string
+}
+
+func (e *invalidSnapshotConfig) Error() string {
+	return fmt.Sprintf("invalid config: %s", e.msg)
+}
+
+func getSnapshotConfig() (snapshotConfig, error) {
+	var conf snapshotConfig
+	conf.timeout = config.Section("Snapshots").Key("timeout_in_seconds").MustInt(60)
```

Comment:
would you prefer strings and use time.ParseDuration()? up to you, int with name of timeout_in_seconds is also not bad assuming this is always a few seconds

Reply:
Think our documentation for this is already written, probably best not to change at this point.